ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,232 @@ namespace ngraph
-                        return node.default_single_output_mapping(
-                            {std::make_shared<ngraph::opset6::MaxPool>(
-                                data,
-                                ngraph::Strides(strides.begin(), strides.end()),
-                                ngraph::Shape(paddings.begin(), paddings.end()),
-                                ngraph::Shape(paddings.begin(), paddings.end()),
-                                ngraph::Shape(kernel_shape.begin(), kernel_shape.end()),
-                                rounding_type)},
-                            {"Out"});
+                        pooling_type = "max";
                     }
-                    else if (pooling_type == "avg" &&
-                             (global_pooling ||
-                              (adaptive && all_of(kernel_shape.begin(),
-                                                  kernel_shape.end(),
-                                                  [](int32_t s) { return s == 1; }))))
+
+                    PDPD_ASSERT((pooling_type == "max") || (pooling_type == "avg"),
+                                "pool2d: not supported pooling type !");
+                    PDPD_ASSERT(kernel_shape.size() == 1 || kernel_shape.size() == 2,
+                                "pool2d: ksize must be 1 or 2!");
+
+                    PartialShape input_shape = data.get_partial_shape();
+                    PDPD_ASSERT(input_shape.rank().is_static(), "pool2d: X rank must be static!");
+                    int32_t input_rank = input_shape.rank().get_length();
+                    PDPD_ASSERT(input_rank >= 2, "kernel rank must be greater than 2");
+                    uint64_t input_h = input_shape[input_rank - 2].get_length();
+                    uint64_t input_w = input_shape[input_rank - 1].get_length();
+
+                    auto auto_pad = ngraph::op::PadType::EXPLICIT;
+                    ngraph::Shape pad_begin, pad_end;
+                    get_paddings(node, pad_begin, pad_end, auto_pad);
+
+                    if (global_pooling ||
+                        (adaptive && std::any_of(kernel_shape.begin(),
+                                                 kernel_shape.end(),
+                                                 [](int32_t i) { return i == 1; })))
                     {
-                        // TODO : resolve axes according to rank
-                        auto axes =
-                            ngraph::opset6::Constant::create(ngraph::element::i64, {2}, {2, 3});
-                        return node.default_single_output_mapping(
-                            {std::make_shared<ngraph::opset6::ReduceMean>(data, axes, true)},
-                            {"Out"});
+                        if (pooling_type == "max")
+                        {
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::MaxPool>(
```

Comment:
This is wrong, you have to use ReduceMax op here

Reply:
@mvafin One question about the ReduceMax, since our Ngraph Op has MaxPool which definition matches with PDPD maxpool here. Why we need ReduceMax to this conversion, is there any limitation of MaxPool ?